### PR TITLE
4.2.0 release

### PR DESCRIPTION
**Pull request by xinlian12**
### New Features
* Added script logging enabled API to `CosmosStoredProcedureRequestOptions`.
* Updated `DirectConnectionConfig` default `idleEndpointTimeout` to 1h and default `connectTimeout` to 5s.
### Key Bug Fixes
* Fixed issue where `GatewayConnectionConfig` `idleConnectionTimeout` was overriding `DirectConnectionConfig` `idleConnectionTimeout`.
* Fixed `responseContinuationTokenLimitInKb` get and set APIs in `CosmosQueryRequestOptions`.
* Fixed issue in query and change feed when recreating the collection with same name.
* Fixed issue with top query throwing ClassCastException.
* Fixed issue with order by query throwing NullPointerException.
* Fixed `Operator called default onErrorDropped` when stream got cancelled.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+### New Features
+* Added script logging enabled API to `CosmosStoredProcedureRequestOptions`.
+* Updated `DirectConnectionConfig` default `idleEndpointTimeout` to 1h and default `connectTimeout` to 5s.
+### Key Bug Fixes
+* Fixed issue where `GatewayConnectionConfig` `idleConnectionTimeout` was overriding `DirectConnectionConfig` `idleConnectionTimeout`.
+* Fixed `responseContinuationTokenLimitInKb` get and set APIs in `CosmosQueryRequestOptions`.
+* Fixed issue in query and change feed when recreating the collection with same name.
+* Fixed issue with top query throwing ClassCastException.
+* Fixed issue with order by query throwing NullPointerException.
+* Fixed `Operator called default onErrorDropped` when stream got cancelled.
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: `* Fixed `Operator called default onErrorDropped` when stream got cancelled.` what does this mean for customer?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: It reflects this fix : https://github.com/Azure/azure-sdk-for-java/pull/12836

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: can this be reworded to be understandable by customer?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: Sure, suggestions ?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: how about `Fixed handling of cancelled requests in the IO stack.` ?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: So this is the original fix from @David-Noble-at-work : 


We now ensure that a pending Direct TCP request in a reactive stream dropped by an end user or the HA layer completes without bubbling up to `reactor.core.publisher.Hooks.onErrorDropped` as a `CompletionException` error. Pending requests may be left outstanding when, for example, an end user calls `CosmosAsyncClient.close` or the HA layer detects that a partition split has occurred. The code now guarantees that each pending `Mono<StoreResponse>` in the stream will run to completion with a new subscriber. Consequently the default `onErrorDropped` handler will not be called thus preventing these distracting error messages.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:13]: @moderakh  - how about this ?

```
Fixed issue in handling of pending requests in direct mode causing Operator called default onErrorDropped being called. 

```

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: it is not the pending requests which cause issue, it is the cancelled requests. no?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: essentially an RNTBD IO request which is cancelled as per request of higher stack will trigger a failure and result in bubbling up a failure from RNTBD to consistency stack

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: As discussed offline, updated to:
* Fixed issue in handling of cancelled requests in direct mode causing reactor `onErrorDropped` being called.

**Pull request merged**